File: Algorithms/Warmup/Easy/Time Conversion/Time Conversion.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Result
{

    /*
     * Complete the 'timeConversion' function below.
     *
     * The function is expected to return a STRING.
     * The function accepts STRING s as parameter.
     */

    public static string timeConversion(string s)
    {
        string hour = s.Substring(0,2);
        string id = s.Substring(8,2);
        int iHour = Int32.Parse(hour);

        if( id == "PM" )
        {
            if( iHour != 12 )
                iHour += 12;

            hour = iHour.ToString();
        }

        if( id == "AM" )
        {
            if( iHour == 12 )
                hour = "00";
            else if( iHour < 10 )
                hour = "0" + iHour.ToString();
            else
                hour = iHour.ToString();
        }

        return hour + s.Substring(2,6);

    }

}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        string s = Console.ReadLine();

        string result = Result.timeConversion(s);

        textWriter.WriteLine(result);

        textWriter.Flush();
        textWriter.Close();
    }
}

File: Algorithms/Warmup/Easy/Time Conversion/Time Conversion.cpp
#include <bits/stdc++.h>

using namespace std;

/*
 * Complete the 'timeConversion' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts STRING s as parameter.
 */

string timeConversion(string s) {
    string hour = s.substr(0,2);
    string id = s.substr(8,2);
    int iHour = stoi(hour.c_str());
    if( id == "PM" ){
        if( iHour != 12 )
           iHour += 12;

        hour = to_string(iHour);
    }

    if( id == "AM" ){
        if( iHour == 12 )
           hour = "00";
        else if( iHour < 10 )
           hour = "0" + to_string(iHour);
        else
           hour = to_string(iHour);
    }

    return hour + s.substr(2,6);
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    string result = timeConversion(s);

    fout << result << "\n";

    fout.close();

    return 0;
}
